### PR TITLE
test-network: add tests for IPv6ProxyNDPAddress=

### PR DESCRIPTION
**Pull request by yuwata**
For #18378.

### CODE REVIEW

File: src/network/networkd-ipv6-proxy-ndp.c [hunk 1]
```diff
@@ -38,7 +38,7 @@ static int ipv6_proxy_ndp_address_configure(Link *link, const struct in6_addr *a
         if (r < 0)
                 return log_link_error_errno(link, r, "Could not create RTM_NEWNEIGH message: %m");
 
-        r = sd_rtnl_message_neigh_set_flags(req, NLM_F_REQUEST | NTF_PROXY);
+        r = sd_rtnl_message_neigh_set_flags(req, NTF_PROXY);
```
Comment (ssahani) [inline src/network/networkd-ipv6-proxy-ndp.c:41]: `NLM_F_REQUEST`  is not associated with ndp. It's must be inn all request messages in netlink
https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/ip/ipneigh.c#n113

File: src/network/networkd-ipv6-proxy-ndp.c [hunk 2]
```diff
@@ -18,7 +18,7 @@ static int set_ipv6_proxy_ndp_address_handler(sd_netlink *rtnl, sd_netlink_messa
         assert(link);
 
         r = sd_netlink_message_get_errno(m);
-        if (r < 0 && r != -EEXIST)
+        if (r < 0)
```
Comment (ssahani) [inline src/network/networkd-ipv6-proxy-ndp.c:21]: If `NLM_F_EXCL` is set then `EEXIST` will come. I think we should better add `NLM_F_EXCL` `sd_rtnl_message_new_neigh` like other which is missing now

Comment (yuwata) [reply src/network/networkd-ipv6-proxy-ndp.c:21]: No. When `NTF_PROXY` is set `pneigh_lookup()` is called, and `EEXIST` never come. There are no cases we should ignore.

File: src/network/networkd-ipv6-proxy-ndp.c (hunk 1 above)
Comment (yuwata) [reply src/network/networkd-ipv6-proxy-ndp.c:41]: `NLM_F_REQUEST` is set by default in `message_new()` which is called in `sd_rtnl_message_new_neigh()`. So, it is not necessary to set explicitly here.

File: src/network/networkd-ipv6-proxy-ndp.c (hunk 2 above)
Comment (ssahani) [reply src/network/networkd-ipv6-proxy-ndp.c:21]: It's not about `NTF_PROXY` only . In general like in _new

File: src/network/networkd-ipv6-proxy-ndp.c (hunk 1 above)
Comment (ssahani) [reply src/network/networkd-ipv6-proxy-ndp.c:41]: Umm then we should fix the commit message according to that

File: src/network/networkd-ipv6-proxy-ndp.c (hunk 2 above)
Comment (yuwata) [reply src/network/networkd-ipv6-proxy-ndp.c:21]: Not sure for general case... It should depend on the object we want to add or update...

File: src/network/networkd-ipv6-proxy-ndp.c (hunk 1 above)
Comment (yuwata) [reply src/network/networkd-ipv6-proxy-ndp.c:41]: Hmm? The problem here is that `NLM_F_REQUEST` is a nlmsg flag and used in `sd_netlink_message_set_flags()`. Should not used in `sd_rtnl_message_neigh_set_flags()`. I already commented so.

Comment (ssahani) [reply src/network/networkd-ipv6-proxy-ndp.c:41]: Yeah that's right

File: src/network/networkd-ipv6-proxy-ndp.c (hunk 2 above)
Comment (ssahani) [reply src/network/networkd-ipv6-proxy-ndp.c:21]: In this case `NTF_PROXY` . But need to consider other cases too

Comment (yuwata) [reply src/network/networkd-ipv6-proxy-ndp.c:21]: If it is worth to update, let's adjust the default flags in another later PR.

Comment (ssahani) [reply src/network/networkd-ipv6-proxy-ndp.c:21]: OK lgtm

**Review by bluca — APPROVED** [review]
LGTM - all comments threads seem concluded, setting green flag

**Comment by kpfleming** [comment]
Just came here to report the bogus `NLM_F_REQUEST` flag being passed and I see that it's already been caught :)

**Pull request merged**